libbeat/common/transport/tlscommon/tls.go

```diff
@@ -187,3 +204,58 @@ func ResolveTLSVersion(v uint16) string {
 func ResolveCipherSuite(cipher uint16) string {
 	return tlsCipherSuite(cipher).String()
 }
+
+// PEMReader allows to read a certificate in PEM format either through the disk or from a string.
+type PEMReader struct {
+	reader io.ReadCloser
+}
+
+// NewPEMReader returns a new PEMReader.
+func NewPEMReader(certificate string) (*PEMReader, error) {
+	if IsPEMString(certificate) {
+		return &PEMReader{reader: ioutil.NopCloser(strings.NewReader(certificate))}, nil
+	}
+
+	r, err := os.Open(certificate)
+	if err != nil {
+		return nil, err
+	}
+	return &PEMReader{reader: &ReadCloser{r}}, nil
+}
+
+// Close closes the target io.ReadCloser.
+func (p *PEMReader) Close() error {
+	return p.reader.Close()
+}
+
+// Read read bytes from the io.ReadCloser.
+func (p *PEMReader) Read(b []byte) (n int, err error) {
+	return p.reader.Read(b)
+}
+
+// ReadCloser wraps a io.Reader into a ReadCloser and call close if available on the target.
+type ReadCloser struct {
+	reader io.Reader
+}
+
+// Read proxy the Read call to wrapped reader.
+func (r *ReadCloser) Read(b []byte) (n int, err error) {
+	return r.reader.Read(b)
+}
+
+// Close closes the wrapped reader if it respond to the Close function.
+func (r *ReadCloser) Close() error {
+	if c, ok := r.reader.(io.Closer); ok {
+		return c.Close()
+	}
+	return nil
+}
+
+// IsPEMString returns true if the provided string match a PEM formatted certificate. try to pem decode to validate.
+func IsPEMString(s string) bool {
+	if block, _ := pem.Decode([]byte(strings.TrimSpace(s))); block != nil {
```

Comment:
could we make this more forgiving in case there is a parse error? If parsing failed we should not "think" it is a path if possible.

Reply:
My initial idea for validation was to check only for BEGIN/END, but that seems a bit too fragile.
Anything else you had in mind?

Reply:
I would think that if the string starts with a `-` then it should be tried as a certificate always and let it fail.